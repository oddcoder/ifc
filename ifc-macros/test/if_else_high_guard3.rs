use ifc_macros::ifc_block;

fn main() {
    ifc_block!{
        let a = 5;
        if a == 5 {
            #[IFC(Low)]
            let x = 5;
            x
        } else {
            let x = 6;
            x
        };
    }
}