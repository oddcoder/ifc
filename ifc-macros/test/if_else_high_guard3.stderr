error: Cannot declare new low variable or assign to low variable in high context
 --> $DIR/if_else_high_guard3.rs:8:13
  |
8 |             let x = 5;
  |             ^^^^^^^^^^
  |
help: High context was created here
 --> $DIR/if_else_high_guard3.rs:6:12
  |
6 |         if a == 5 {
  |            ^^^^^^
help: Low variable is used here.
 --> $DIR/if_else_high_guard3.rs:8:17
  |
8 |             let x = 5;
  |                 ^
